libnopegl/src/scene.c

```diff
@@ -150,6 +152,10 @@ static int setup_nodes(void *user_arg, struct ngl_node *parent, struct ngl_node
     if (parent) {
         if (!ngli_darray_push(&parent->children, &node))
             return NGL_ERROR_MEMORY;
+        if (node->cls->draw) {
+            if (!ngli_darray_push(&parent->draw_children, &node))
+                return NGL_ERROR_MEMORY;
+        }
```

Comment:
Why not merge the 2 conditions?

Reply:
Personal preference, I think it is easier to read.